service/api/service.go

```diff
@@ -454,19 +462,23 @@ func (s *serviceImpl) tryPrepareRPCbyDescribe(ctx context.Context, req iwfidl.Wo
 	var searchAttributes []iwfidl.SearchAttribute
 	var dataAttributes []iwfidl.KeyValue
 
-	requestedSAs := req.SearchAttributes
+	var requestedSAs []iwfidl.SearchAttributeKeyAndType
 	saPolicy := req.GetSearchAttributesLoadingPolicy()
-	if saPolicy.GetPersistenceLoadingType() != iwfidl.ALL_WITHOUT_LOCKING {
+	switch saPolicy.GetPersistenceLoadingType() {
+	case iwfidl.PARTIAL_WITH_EXCLUSIVE_LOCK, iwfidl.PARTIAL_WITHOUT_LOCKING:
 		requestedSAKeys := map[string]bool{}
 		for _, saKey := range saPolicy.PartialLoadingKeys {
 			requestedSAKeys[saKey] = true
 		}
-		requestedSAs = []iwfidl.SearchAttributeKeyAndType{}
```

Comment:
Removing this line is the reason that the test fail. 
In the original code, `requestedSAs` is reset here to empty, so that it will put the actual requestedSA into the list, based on the request.